python/py-pyface/Portfile

```diff
@@ -21,38 +21,29 @@ long_description        The pyface project contains a toolkit-independent GUI\
                         abstraction layer, which is used to support the\
                         "visualization" features of the Traits package.
 
-checksums               rmd160  57356c20e10291efe49f58b80d7f5483dad741da \
-                        sha256  d04d1869ec0b63966dd0a64b83fcea15a001a3220cf964e82f2745ad330e37c5 \
-                        size    7756962
+checksums               rmd160  09b4f56cfbb3fd62111ec2c62c675b67eae385bd \
+                        sha256  e34f8fb285517e25bfe41a4b8d1d54860d2636bd5fa5b9bff26ab96c1a098895 \
+                        size    7850673
 
-python.versions         38 39
+python.versions         39 310 311 312
 
 if {${name} ne ${subport}} {
     depends_lib-append      port:py${python.version}-traits \
                             port:py${python.version}-importlib-metadata \
                             port:py${python.version}-pygments
 
-    if {${python.version} < 39} {
-        depends_lib-append  port:py${python.version}-importlib-resources
-    }
-
-    variant pyqt4 description "Qt backend using PyQt4" {
-        depends_lib-append  port:py${python.version}-pyqt4
-    }
```

Comment:
@jjstickel does upstream no longer support PyQt4? If they still do then I can already see @barracuda156 his request coming to keep it alive so that it can be used on older OSes that do not support Qt5. If support has been dropped upstream then there isn't much we can do about it aside from pegging it to an earlier version that still does - but I would not be in favor of that.

Reply:
From the pyface readme (https://github.com/enthought/pyface):

"The following GUI backends are supported:

    PySide2 (stable) and PySide6 (experimental)
    PyQt5 (stable) and PyQt6 (in development)
    wxPython 4 (experimental)"

So, I took that to mean that QT4 was no longer a supported backend. I chose not to include (and test) pyside6, pyqt6, and wxpython-4.

Reply:
@jjstickel Then we should peg the version to the last qt4-supporting system where qt5+ does not work.
At least if GTK backend does not exist (gtk works fine on all systems).

Reply:
I'm not interested in doing the work for this. For context, I doubt many people use these Enthought ports, and they can now be installed via pip. I was on the verge of abandoning them, but decided to keep things going for this round of updates.

Maybe pyside2 works on the legacy systems?

Reply:
@jjstickel It does not really require much work, since the only thing needed is to retain what already exists, not write or tests anything extra, just a matter of a conditional block in the portfile.

But if it is troublesome, I can do it, so that you can just copy-paste from my commit into your PR.

Reply:
We do not need Qt4 variant in fact. We can just use Qt4 on legacy systems by default.

I will handle this now.